hv.c

```diff
@@ -728,12 +722,10 @@ Perl_hv_common(pTHX_ HV *hv, SV *keysv, const char *key, STRLEN klen,
                 }
                 else {
                     /* Effectively this is save_hek_flags() for a new version
-                       of the HEK and Safefree() of the old rolled together.
-                       "masked_flags" doesn't have HVhek_NOTSHARED set.
-                       We could be clearer on this. */
-                    HeKFLAGS(entry) = masked_flags | HVhek_NOTSHARED;
+                       of the HEK and Safefree() of the old rolled together. */
+                    HeKFLAGS(entry) ^= HVhek_WASUTF8;
```

Comment:
Does this mean we previously would have cleared `HVhek_{FREEKEY,PLACEHOLD,KEYCANONICAL}` but now we don't? What is the implication of that?

Reply:
What year was this again? :p

I'm guessing a bit here, not helped by github having lost the commit and I suspect corrupted the context of the conversation, but the original question was trying to understand correctness (not per se complaining about clarity). The possibility that the mentioned flags would simply not fit in `HeKFLAGS(entry)` had not occurred to me, so I'm pretty sure that answers my original question, and I'll mark it resolved.